File: source/SharpFlame/clsGeneratorTileset.cs
using System;

namespace SharpFlame
{
    public class clsGeneratorTileset
    {
        public clsTileset Tileset;

        public struct sUnitChance
        {
            public clsUnitType Type;
            public UInt32 Chance;

            public sUnitChance(clsUnitType Type, UInt32 Chance)
            {
                this.Type = Type;
                this.Chance = Chance;
            }
        }

        public sUnitChance[] ScatteredUnits = new sUnitChance[0];
        public int ScatteredUnitCount;
        public int ScatteredUnitChanceTotal;
        public sUnitChance[] ClusteredUnits = new sUnitChance[0];
        public int ClusteredUnitCount;
        public int ClusteredUnitChanceTotal;

        public int BorderTextureNum = -1;

        public modProgram.sLayerList OldTextureLayers;

        public void ScatteredUnit_Add(sUnitChance NewUnit)
        {
            if ( NewUnit.Type == null )
            {
                return;
            }

            ScatteredUnitChanceTotal += Convert.ToInt32(NewUnit.Chance);

            Array.Resize(ref ScatteredUnits, ScatteredUnitCount + 1);
            ScatteredUnits[ScatteredUnitCount] = NewUnit;
            ScatteredUnitCount++;
        }

        public void ClusteredUnit_Add(sUnitChance NewUnit)
        {
            if ( NewUnit.Type == null )
            {
                return;
            }

            ClusteredUnitChanceTotal += Convert.ToInt32(NewUnit.Chance);

            Array.Resize(ref ClusteredUnits, ClusteredUnitCount + 1);
            ClusteredUnits[ClusteredUnitCount] = NewUnit;
            ClusteredUnitCount++;
        }
    }
}

File: reference/FlaMEVB/clsGeneratorTileset.vb

Public Class clsGeneratorTileset

    Public Tileset As clsTileset

    Public Structure sUnitChance
        Public Type As clsUnitType
        Public Chance As UInteger

        Public Sub New(Type As clsUnitType, Chance As UInteger)

            Me.Type = Type
            Me.Chance = Chance
        End Sub
    End Structure
    Public ScatteredUnits(-1) As sUnitChance
    Public ScatteredUnitCount As Integer
    Public ScatteredUnitChanceTotal As Integer
    Public ClusteredUnits(-1) As sUnitChance
    Public ClusteredUnitCount As Integer
    Public ClusteredUnitChanceTotal As Integer

    Public BorderTextureNum As Integer = -1

    Public OldTextureLayers As sLayerList

    Public Sub ScatteredUnit_Add(NewUnit As sUnitChance)

        If NewUnit.Type Is Nothing Then
            Exit Sub
        End If

        ScatteredUnitChanceTotal += CInt(NewUnit.Chance)

        ReDim Preserve ScatteredUnits(ScatteredUnitCount)
        ScatteredUnits(ScatteredUnitCount) = NewUnit
        ScatteredUnitCount += 1
    End Sub

    Public Sub ClusteredUnit_Add(NewUnit As sUnitChance)

        If NewUnit.Type Is Nothing Then
            Exit Sub
        End If

        ClusteredUnitChanceTotal += CInt(NewUnit.Chance)

        ReDim Preserve ClusteredUnits(ClusteredUnitCount)
        ClusteredUnits(ClusteredUnitCount) = NewUnit
        ClusteredUnitCount += 1
    End Sub
End Class
